vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -0,0 +1,132 @@
+import enum
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+import torch
+from torch.nn.parameter import Parameter
+
+from vllm.model_executor.layers.linear import (LinearMethodBase,
+                                               set_weight_attrs)
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+
+
+class FP8Config(QuantizationConfig):
+    """Config class for FP8."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "fp8"
+
+    @classmethod
+    def get_supported_act_dtypes(cls) -> List[torch.dtype]:
+        return [
+            torch.bfloat16, torch.half, torch.float8_e4m3fn, torch.float8_e5m2
```

Comment:
I don't think `torch.float8_e4m3fn, torch.float8_e5m2` will work at the moment if passed into `apply_weights` since it will always quantize. We would also need to know the `scale`, so a refactor is needed if we want to support that input